src/dbally/_collection.py

```diff
@@ -226,3 +228,29 @@ async def ask(self, question: str, dry_run: bool = False, return_natural_respons
         await event_tracker.request_end(RequestEnd(result=result))
 
         return result
+
+    def get_similarity_indexes(self) -> Dict[AbstractSimilarityIndex, List[Tuple[str, str, str]]]:
+        """
+        List all similarity indexes from all views in the collection.
+
+        Returns:
+            Dictionary with similarity indexes as keys and values containing lists of places where they are used
+            (represented by a tupple containing view name, method name and argument name)
+        """
+        indexes: Dict[AbstractSimilarityIndex, List[Tuple[str, str, str]]] = {}
+        for view_name in self._views:
+            view = self.get(view_name)
+            filters = view.list_filters()
+            for filter_ in filters:
+                for param in filter_.parameters:
+                    if param.similarity_index:
+                        indexes.setdefault(param.similarity_index, []).append((view_name, filter_.name, param.name))
+        return indexes
+
+    async def update_similarity_indexes(self) -> None:
+        """
+        Update all similarity indexes from all views in the collection.
+        """
+        indexes = self.get_similarity_indexes()
+        update_corutines = [index.update() for index in indexes]
+        await asyncio.gather(*update_corutines)
```

Comment:
Hmm, in the case that any update task fails, we will terminate the others, which may cause an unexpected state in the indexes. I think we should use asyncio.gather with `return_exceptions=True` and return some sort of object from `update_similarity_indexes` with an update status or exception.

Reply:
Good catch, will do!